lighthouse-core/audits/dobetterweb/uses-http2.js

```diff
@@ -43,67 +58,174 @@ class UsesHTTP2Audit extends Audit {
     return {
       id: 'uses-http2',
       title: str_(UIStrings.title),
-      failureTitle: str_(UIStrings.failureTitle),
       description: str_(UIStrings.description),
-      requiredArtifacts: ['URL', 'devtoolsLogs'],
+      requiredArtifacts: ['URL', 'devtoolsLogs', 'traces'],
     };
   }
 
+  /**
+   * Computes the estimated effect all results being converted to use http/2, the max of:
+   *
+   * - end time of the last long task in the provided graph
+   * - end time of the last node in the graph
+   *
+   * @param {Array<{url: string}>} results
+   * @param {Node} graph
+   * @param {Simulator} simulator
+   * @return {number}
+   */
+  static computeWasteWithTTIGraph(results, graph, simulator) {
+    const beforeLabel = `uses-http2-before`;
+    const afterLabel = `uses-http2-after`;
+    const flexibleOrdering = true;
+
+    const urlsToChange = new Set(results.map(result => result.url));
+    const simulationBefore = simulator.simulate(graph, {label: beforeLabel, flexibleOrdering});
+
+    // Update all the protocols to reflect implementing our recommendations
+    /** @type {Map<string, string>} */
+    const originalProtocols = new Map();
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      if (!urlsToChange.has(node.record.url)) return;
+
+      originalProtocols.set(node.record.requestId, node.record.protocol);
+      node.record.protocol = 'h2';
+    });
+
+    const simulationAfter = simulator.simulate(graph, {label: afterLabel, flexibleOrdering});
+
+    // Restore the original protocol after we've done our simulation
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      const originalProtocol = originalProtocols.get(node.record.requestId);
+      if (originalProtocol === undefined) return;
+      node.record.protocol = originalProtocol;
+    });
+
+    const savingsOnOverallLoad = simulationBefore.timeInMs - simulationAfter.timeInMs;
+    const savingsOnTTI = Interactive.getLastLongTaskEndTime(simulationBefore.nodeTimings) -
+      Interactive.getLastLongTaskEndTime(simulationAfter.nodeTimings);
+    const savings = Math.max(savingsOnTTI, savingsOnOverallLoad);
+
+    // Round waste to nearest 10ms
+    return Math.round(Math.max(savings, 0) / 10) * 10;
```

Comment:
aside: I wonder if we should drop this rounding and just rely on `granularity` in the report view.

Reply:
wdyt?

Reply:
oh sorry, uh no strong opinion, but I lean to keeping the rounding. 

I like the idea that programmatic users aren't misled with the level of precision here (which relying on default granularity would obscure), but I won't die on that hill :)

Reply:
My issue is that 10 is rather arbitrary ... why not 20? 50? 100?

Reply:
> leave precision as an exercise to the consumer

If our consumers were as knowledgable about the inherent variability and *lack* of precision our numbers typically have I would love to do that (and have believed it in the past).